pkg/environment/environment.go

```diff
@@ -288,9 +287,19 @@ func GetSubstrate() (substrate.Manager, error) {
 		return nil, errors.Wrap(err, "failed to get boot environment")
 	}
 
-	poolOnce.Do(func() {
+	slices.Sort(subURLs)
+	slices.Sort(env.SubstrateURL)
+
+	// if substrate url changed then update subURLs and update pool with new manager
+	if !slices.Equal(subURLs, env.SubstrateURL) {
+		log.Debug().Strs("substrate_urls", env.SubstrateURL).Msg("updating to sub manager with url")
+		subURLs = env.SubstrateURL
 		pool = substrate.NewManager(env.SubstrateURL...)
```

Comment:
here u are creating connection everytime getSubstrate is called which is against the original design  this should be executed one time better to have another approach than this we should be using one connection all the time, so either we detect if connection is not working we restart the node services if we can reach substrate on another url from zos config something like monitoring